source/extensions/common/wasm/context.cc

```diff
@@ -1321,6 +1322,8 @@ Network::FilterStatus convertNetworkFilterStatus(proxy_wasm::FilterStatus status
     return Network::FilterStatus::Continue;
   case proxy_wasm::FilterStatus::StopIteration:
     return Network::FilterStatus::StopIteration;
+  default:
```

Comment:
I am not sure why we want to do this.  Not having default causes a compilation error if the enum changes since it requires the case to be exhaustive.

Reply:
Not having a default case results in compiler error under gcc. It's either that, or each of those methods needs to have an explicit return at the end.

Reply:
Ah.  Haven't used gcc in a while and a vaguely remember it can change with optimization level as well.  Let me see if I can reproduce it.   What version of gcc are you using?

Reply:
Please take a look at https://github.com/jplevyak/envoy-wasm/tree/wasm-case
where I added the NOT_IMPLEMENTED_GCOVR_EXCL_LINE outside the case and
fixed the handling of  spdlog::level::off:.

I think this addresses all the issues.  WDYT?

Reply:
To be honest, I think that `NOT_IMPLEMENTED_GCOVR_EXCL_LINE` inside `switch` as a `default` case is technically a bit more correct than outside of it, even though it results in effectively the same code.

Reply:
The problem with adding a default statement is that it prevents the compiler from issuing a warning when the cases are not exhaustive such as when new elements are added to the enum.  See https://abseil.io/tips/147   The suggested solution is the one that I am providing: to not have a default but to handle those cases outside the switch.  In this case, the range of the enum is validated in the exports.cc file for all cases, however, that isn't something which is manifest in the C++ type system, so we need to handle non-enum values somehere.

Reply:
Fair point.